### PR TITLE
chore - update npm dependencies and devDependencies

### PR DESCRIPTION
**Pull request by devinea**

### CODE REVIEW

File: packages/ui5-config/package.json [hunk 1]
```diff
@@ -21,12 +21,12 @@
...
     "dependencies": {
-        "@sap-ux/yaml": "workspace:*",
-        "typescript": "4.2.4"
+        "@sap-ux/yaml": "workspace:*"
     },
     "devDependencies": {
         "mem-fs": "2.1.0",
-        "mem-fs-editor": "9.0.0"
+        "mem-fs-editor": "9.0.0",
+        "typescript": "4.2.4"
```
Comment (devinea) [inline packages/ui5-config/package.json:29]: should not have been a dependency!!

File: packages/ui5-config/src/types.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export interface ProxyUIConfig {
     directLoad?: boolean;
 }
 
-export interface CustomMiddleware<C extends any> {
+export interface CustomMiddleware<C> {
```
Comment (devinea) [inline packages/ui5-config/src/types.ts:22]: fix eslint error with newer eslint version
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-type-constraint.md

File: packages/ui5-application-writer/package.json [hunk 3]
```diff
@@ -27,11 +27,11 @@
         "json-merger": "1.1.6"
     },
     "devDependencies": {
-        "@types/ejs": "3.0.6",
+        "@types/ejs": "3.1.0",
         "@types/mem-fs": "1.1.2",
         "@types/fs-extra": "9.0.13",
         "@types/mem-fs-editor": "7.0.0",
-        "typescript": "4.2.4",
+        "typescript": "4.4.4",
```
Comment (kranthie-sap) [inline packages/ui5-application-writer/package.json:34]: @devinea: Is it not sufficient to have this at the root?

**Review by kranthie-sap — CHANGES_REQUESTED** [review]

File: packages/ui5-application-writer/package.json (hunk 3 above)
Comment (devinea) [reply packages/ui5-application-writer/package.json]: Should be, don't recall why they were added in the packages. Removed in https://github.com/SAP/open-ux-tools/pull/163/commits/24d69e71333a00927b1bff619e79e6e1a7875531

**Review by tobiasqueck — APPROVED** [review]

**Review by kranthie-sap — APPROVED** [review]
Looks good to me.

**Pull request merged**